src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs

```diff
@@ -8411,6 +8411,104 @@ public static int Main()
+            comp.VerifyDiagnostics();
+
+            var wellKnownType = comp.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_SkipLocalsInitAttribute);
+            var typeMember = comp.GetMember<NamedTypeSymbol>("System.Runtime.CompilerServices.SkipLocalsInitAttribute");
+
+            Assert.Equal(wellKnownType, typeMember);
+        }
+
+        [Fact]
+        public void SkipLocalsInitAttributeRecognizedAsWellKnownTypeWhenDeclaredInMetadata()
+        {
+            var source = @"
+namespace System.Runtime.CompilerServices
+{
+    public class SkipLocalsInitAttribute : System.Attribute
+    {
+    }
+}
+";
+
+            var metadataComp = CreateStandardCompilation(source);
+            metadataComp.VerifyDiagnostics();
+
+            var comp = CreateStandardCompilation("", references: new[] { metadataComp.EmitToImageReference() });
+            comp.VerifyDiagnostics();
+
+            var wellKnownType = comp.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_SkipLocalsInitAttribute);
+            var typeMember = comp.GetMember<NamedTypeSymbol>("System.Runtime.CompilerServices.SkipLocalsInitAttribute");
+
+            Assert.Equal(wellKnownType, typeMember);
+        }
+
+        [Fact]
+        public void SkipLocalsInitAttributeFromSourceRecognizedAsWellKnownTypeWhenDeclaredInSourceAndMetadata()
+        {
+            var source = @"
+namespace System.Runtime.CompilerServices
+{
+    public class SkipLocalsInitAttribute : System.Attribute
+    {
+    }
+}
+";
+
+            var metadataComp = CreateStandardCompilation(source);
+            metadataComp.VerifyDiagnostics();
+
+            var comp = CreateStandardCompilation(source, references: new[] { metadataComp.EmitToImageReference() });
+            comp.VerifyDiagnostics();
+
+            var wellKnownType = comp.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_SkipLocalsInitAttribute);
+            
+            Assert.IsType<SourceNamedTypeSymbol>(wellKnownType);
+        }
+
+        [Fact]
+        public void SkipLocalsInitAttributeMissingWhenDeclaredMoreThanOnceInOtherAssemblies()
+        {
+            var source = @"
+namespace System.Runtime.CompilerServices
+{
+    public class SkipLocalsInitAttribute : System.Attribute
+    {
+    }
+}
+";
+
+            var metadataComp1 = CreateStandardCompilation(source);
+            metadataComp1.VerifyDiagnostics();
+
+            var metadataComp2 = CreateStandardCompilation(source);
+            metadataComp2.VerifyDiagnostics();
+
+            var comp = CreateStandardCompilation("", references: new[] { metadataComp1.EmitToImageReference(), metadataComp2.EmitToImageReference() });
+            comp.VerifyDiagnostics();
+
+            var wellKnownType = comp.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_SkipLocalsInitAttribute);
+
+            Assert.True(wellKnownType.IsErrorType());
+        }
+
+        #endregion
```

Comment:
Couple of other tests we should add:

- Negative test: trying to find the member when it is not deifned 
- What happens when it does not dreive from Attribute? Does that matter or not?

Reply:
Negative test added.

Reply:
It actually doesn't matter if it's derived from Attribute or not. We are only testing whether it is being properly recognized as a Well Known Type.

Reply:
I think this is the right behavior. If it doesn't inherit from Attribute you won't be able to use it anywhere and it won't mean anything, but I think we should still consider the type well-known.